Automattic/jetpack
# Boost: Use "critical css gen" package from the monorepo

Fixes https://github.com/Automattic/jetpack/issues/38584.

There is however a problem. If you want to check something you're importing from the package and you cmd + click on the package name when importing it in Boost (for example [this](https://github.com/Automattic/jetpack/blob/7cba6776d5d86b687b0c17bdd968c3d88f5b8f63/projects/plugins/boost/app/assets/src/js/features/critical-css/lib/generate-critical-css.ts#L9)), it opens the `node.ts` instead of `browser.ts`. Because of that, the `BrowserInterfaceIframe` reference is not found in [`global.d.ts`](https://github.com/Automattic/jetpack/blob/7cba6776d5d86b687b0c17bdd968c3d88f5b8f63/projects/plugins/boost/app/assets/src/js/global.d.ts#L47).

This is probably a package issue, but I don't know how to properly fix this.

## Proposed changes:

* Update Boost to use critical css gen package from the monorepo, instead of the one from [the external repo](https://github.com/Automattic/jetpack-boost-critical-css-gen).

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

- Setup Boost (free);
- Make sure Critical CSS generation works properly.
